Add option to show selected labels in the issue line

aih opened the pull request:
This adds an option, `--issue-line-labels`, which allows the user to list the specified Github labels, in brackets, next to each issue.

For example, when issue #367 of this repository is closed, the Changelog for that issue, with option  `--issue-line-labels=ALL` will look like this: 

- Feature Request: On using a public token on a private repo and getting a 404, fail informatively #367 [feature-request][first-timers-only][help-wanted]

Selected labels can be specified in the option:
`--issue-line-labels=first-timers-only` will generate the following line:
- Feature Request: On using a public token on a private repo and getting a 404, fail informatively #367 [first-timers-only]

Related to #316

olleolleolle commented:
Hi Ari!

Glad you wanted to add a feature to this tool. 

To make this PR mergeable, you need to rebase it on top of master. At this point, master has OctoKit in it.

- To make the feature more useful to users, it needs **more description**, and I suppose the Description field of this PR is a good place to begin. Add usage and a few examples of what it looks like.
- The **name of the new option** has "show" in it, which does not fit perfectly with the list of label names that are to be filtered-and-suffixed onto the issue line. ("Show" implies a Boolean "on" setting.) Perhaps something like `--issue-line-labels=First-Timers-Only,Documentation,CssChanges`?

It's awesome to see new people modify this to their needs. Can you find a way to make the above happen? If not, would you want me to make the changes, carrying your PR into master?

aih commented:
I've updated the code and rebased on top of the latest Master commit. I can also update the [wiki page](https://github.com/skywinder/github-changelog-generator/wiki/Advanced-change-log-generation-examples) to document this option once the pull request is merged. I've found this to be a very flexible and helpful utility for creating Changelogs. 
I have integrated it with Grunt.js and convert the .md to html for my application's webpage.

crtrott commented:
The second feature I was thinking of as missing after using the tool for the first time 👍

olleolleolle commented on lib/github_changelog_generator/parser.rb:
```diff
@@ -205,6 +208,7 @@ def self.default_options
         enhancement_labels: %w(enhancement Enhancement),
         bug_labels: %w(bug Bug),
         exclude_labels: %w(duplicate question invalid wontfix Duplicate Question Invalid Wontfix),
+        issue_line_labels: [],
```
The `KNOWN_OPTIONS` constant needs this setting name as well:

https://github.com/skywinder/github-changelog-generator/blob/master/lib/github_changelog_generator/options.rb#L7

olleolleolle commented on lib/github_changelog_generator/generator/generator_generation.rb:
```diff
@@ -143,13 +143,27 @@ def get_string_for_issue(issue)
+               end
+      labels_string = labels.map { |label| " \[[#{label['name']}](#{label['url'].sub! 'api.github.com/repos', 'github.com'})\]" }.join("")
+      line = "#{line}#{labels_string}"
+      line
```
I would be more comfortable with a function that only calculated the labels string from the `issue` data:

```ruby
def line_labels_for(issue)
  labels = if options[:issue_line_labels] == ["ALL"]
             issue["labels"]
           else
             issue["labels"].select { |label| options[:issue_line_labels].include?(label["name"]) }
           end
  labels.map { |label| " \[[#{label['name']}](#{label['url'].sub('api.github.com/repos', 'github.com')})\]" }.join("")
end
```

This could be used like

```ruby
if options[:issue_line_labels].present?
  title_with_number = "#{title_with_number}{line_labels_for(issue)}"
end
```

aih replied:
@olleolleolle, this all works for me. Thanks for making the code changes (I'm not a Ruby developer, but this project has been very useful). I've updated the code for this issue and rebased.

olleolleolle commented:
@aih In my changes, I used ActiveSupport core_ext methods. That's not prevalent around all of the code, but it's there.

olleolleolle commented:
Thanks, @aih! This looks grand. I'll merge it, pending all the builds. Yay!

olleolleolle commented:
:party:

skywinder commented:
Great addition! Thanks! 👍

skywinder commented on lib/github_changelog_generator/generator/generator_generation.rb:
```diff
@@ -143,11 +143,23 @@ def get_string_for_issue(issue)
+               end
+      labels.map { |label| " \[[#{label['name']}](#{label['url'].sub('api.github.com/repos', 'github.com')})\]" }.join("")
+    end
+
```
@aih can you add tests to covers cases with new method?
